File: src/jlppc/regimys/evolution/E_Level.java
package jlppc.regimys.evolution;

import jlppc.regimys.objects.Espece;
import jlppc.regimys.objects.Pokemon;

public class E_Level extends Evolution {
	
	protected int level;
	
	public E_Level(int evo, int level) {
		super(evo);
		this.level = level;
	}

	@Override
	public boolean checkEvolve(Pokemon toCheck) {
		return (toCheck.getLevel() >= level);
	}
}

File: src/jlppc/regimys/evolution/ELevel.cpp
#include "evolutions.hpp"


namespace Evolutions {
	E_Level::E_Level(int evo, int level) : Evolution(evo) {
		this->level = level;
	}

	bool E_Level::checkEvolve(OpMon const &toCheck) const {
		return (toCheck.getLevel() >= level);
	}
}


